From f45500823bd3fa10ae7099209c7b338139ec501d Mon Sep 17 00:00:00 2001
From: Divya Ranjan <divya@subvertising.org>
Date: Wed, 11 Dec 2024 01:15:10 +0000
Subject: [PATCH 09/10] gnu: Add rust-gtk4-sys-0.4.

* gnu/packages/crates-gtk.scm (rust-gtk4-sys-0.4): New variable.

Change-Id: If6ca197a6ee60044f010155181cdea499d520d3e
---
 gnu/packages/crates-gtk.scm | 33 +++++++++++++++++++++++++++++++++
 1 file changed, 33 insertions(+)

diff --git a/gnu/packages/crates-gtk.scm b/gnu/packages/crates-gtk.scm
index 7272c55689..026d0bc518 100644
--- a/gnu/packages/crates-gtk.scm
+++ b/gnu/packages/crates-gtk.scm
@@ -4370,6 +4370,39 @@ (define-public rust-gtk4-sys-0.7
        #:cargo-development-inputs (("rust-shell-words" ,rust-shell-words-1)
                                    ("rust-tempfile" ,rust-tempfile-3))))))
 
+(define-public rust-gtk4-sys-0.4
+  (package
+    (name "rust-gtk4-sys")
+    (version "0.4.8")
+    (source
+      (origin
+        (method url-fetch)
+        (uri (crate-uri "gtk4-sys" version))
+        (file-name
+          (string-append name "-" version ".tar.gz"))
+        (sha256
+          (base32
+            "0qqgxfbmygsl3xd3qal37cdz4ibfc0j9xxrzv9r7qjv3x9p01j2v"))))
+    (build-system cargo-build-system)
+    (arguments
+      `(#:cargo-inputs
+        (("rust-cairo-sys-rs" ,rust-cairo-sys-rs-0.20)
+         ("rust-gdk-pixbuf-sys" ,rust-gdk-pixbuf-sys-0.15)
+         ("rust-gdk4-sys" ,rust-gdk4-sys-0.4)
+         ("rust-gio-sys" ,rust-gio-sys-0.15)
+         ("rust-glib-sys" ,rust-glib-sys-0.15)
+         ("rust-gobject-sys" ,rust-gobject-sys-0.15)
+         ("rust-graphene-sys" ,rust-graphene-sys-0.15)
+         ("rust-gsk4-sys" ,rust-gsk4-sys-0.4)
+         ("rust-libc" ,rust-libc-0.2)
+         ("rust-pango-sys" ,rust-pango-sys-0.15)
+         ("rust-system-deps" ,rust-system-deps-6))))
+    (home-page "https://gtk-rs.org/gtk4-rs")
+    (synopsis "FFI bindings of GTK 4")
+    (description
+      "This package provides FFI bindings of GTK 4.")
+    (license license:expat)))
+
 (define-public rust-libadwaita-0.6
   (package
     (name "rust-libadwaita")
-- 
2.46.0

